src/registers/model_specific.rs

```diff
@@ -445,4 +474,35 @@ mod x86_64 {
             unsafe { msr.write(value.bits()) };
         }
     }
+
+    impl UCet {
+        /// Read the raw IA32_U_CET.
+        #[inline]
+        pub fn read_raw() -> u64 {
+            unsafe { Self::MSR.read() }
+        }
+
+        /// Write the raw IA32_U_CET.
+        #[inline]
+        pub fn write_raw(value: u64) {
```

Comment:
`write_raw()` is not marked unsafe because Rust's unsafe is only related to check memory de-referencing by the compiler and the type system. E.g. `MSR::write()` is unsafe because you can give it incorrect MSR address but actually `Efer::write() is not unsafe. Disabling long mode is a legit action with collateral consequences.

Reply:
> [...] but actually `Efer::write()` is not unsafe. Disabling long mode is a legit action with collateral consequences.

Disabling long mode could cause the following the code to be interpreted as 32bit instructions which could do a lot of unexpected things such as reading and writing to memory.